### ISSUE TITLE
Failed to compile - annot find function `list_dialog` in crate `tinyfiledialogs`

### ISSUE THREAD

**Issue by Monochrome-Sauce** [servo/servo#31205]
I followed the instructions at https://github.com/servo/servo#Linux for my distro (debian based), everything went fine afaik.

Running `./mach build -d` fails with the following error:
```rs
error[E0425]: cannot find function `list_dialog` in crate `tinyfiledialogs`
   --> ports/servoshell/webview.rs:618:36
    |
618 |             match tinyfiledialogs::list_dialog("Choose a device", &["Id", "Name"], dialog_rows) {
    |                                    ^^^^^^^^^^^ not found in `tinyfiledialogs`
```
I'm new to Rust and Cargo, but to me it seems like the `ports/servoshell/Cargo.toml` file uses the default crates.io version rather than the Servo project's version?

**Comment by Taym95** [comment]
@Monochrome-Sauce  can you change `tinyfiledialogs = "3.0"` to `tinyfiledialogs = { workspace = true }` and try?

**Comment by Taym95** [comment]
> @Monochrome-Sauce can you change `tinyfiledialogs = "3.0"` to `tinyfiledialogs = { workspace = true }` and try?

I don't think this will help, I see [list_dialog was removed](https://github.com/jdm/tinyfiledialogs-rs/commit/b982cc93862b743769725c7eb33f4635793a9b37) in `3.9.0` but still used servo!

**Comment by Monochrome-Sauce** [comment]
> @Monochrome-Sauce can you change `tinyfiledialogs = "3.0"` to `tinyfiledialogs = { workspace = true }` and try?

It didn't work.
```
error: failed to parse manifest at `ports/servoshell/Cargo.toml`

Caused by:
  error inheriting `tinyfiledialogs` from workspace root manifest's `workspace.dependencies.tinyfiledialogs`

Caused by:
  `dependency.tinyfiledialogs` was not found in `workspace.dependencies`
```

**Comment by mrobinson** [comment]
@Monochrome-Sauce Have you made any changes to the `Cargo.lock` file? It seems like you are trying to compile Servo with a newer version of `tinyfiledialogs`.

In the future, we should probably replace tinyfiledialogs with egui-modal or something similar.

**Comment by vareille** [comment]
this function in tinyfiledialogs was moved to an additionnal file called
more_dialogs/tinyfd_modeialogs.c
